cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/test/java/io/cdap/cdap/datapipeline/DraftServiceTest.java

```diff
@@ -41,19 +41,27 @@
 import io.cdap.cdap.internal.guava.reflect.TypeToken;
```

Comment:
Where is the change in DraftService.java? 

Also could we move the auth check from DraftService to DraftHandler ? Existing auth enforcement in appfab is done in handlers instead of service, as that allows service APIs to used in cases where there is no associated user requests.

Reply:
It's not a change in DraftService. The actual check is done when DraftHandler calls `io.cdap.cdap.datapipeline.service.AbstractDataPipelineHandler#respond` and then `io.cdap.cdap.api.Admin#getNamespaceSummary`  to construct NamespaceSummary. Due to an security handler installed the thread now has user principal and uses that to perform enforcement.
So, no enforcement in service, everything is in handler.

Reply:
Sounds good. So now we are piggybacking on namespaceQueryAdmin calls for namespace auth enforcement. 

1) For draft create/update, it will be allowed  if having namespace.get permission ? 
2) Does namespace metadata get request to appfab contain user's credential in auth header ? 
From a quick look, NamespaceQueryAdmin seems to be bound to RemoteNamespaceQueryClient.
It seems to only set user id as header when talking to appfab. The EUC is not set.

Reply:
Rest of PR looks good to me. Feel free to address above comments in a follow-up PR.

Reply:
1. yes. Ia m going to have a second PR with more specific enforcement. As of now it's just namespace access.
2. I'll test it